pkg/boot/netboot/ipxe/ipxe.go

```diff
@@ -69,19 +78,51 @@ func (c *parser) getAndParseFile(u *url.URL) error {
 	if !strings.HasPrefix(config, "#!ipxe") {
 		return ErrNotIpxeScript
 	}
-	log.Printf("Got ipxe config file %s:\n%s\n", r, config)
+	c.log.Printf("Got ipxe config file %s:\n%s\n", r, config)
+
+	// Parent dir of the config file.
+	c.wd = &url.URL{
+		Scheme: u.Scheme,
+		Host:   u.Host,
+		Path:   path.Dir(u.Path),
+	}
 	return c.parseIpxe(config)
 }
 
 // getFile parses `surl` and returns an io.Reader for the requested url.
 func (c *parser) getFile(surl string) (io.ReaderAt, error) {
-	u, err := url.Parse(surl)
+	u, err := parseURL(surl, c.wd)
 	if err != nil {
 		return nil, fmt.Errorf("could not parse URL %q: %v", surl, err)
 	}
 	return c.schemes.LazyFetch(u)
 }
 
+func parseURL(name string, wd *url.URL) (*url.URL, error) {
+	u, err := url.Parse(name)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse URL %q: %v", name, err)
+	}
+
+	// If it parsed, but it didn't have a Scheme or Host, use the working
+	// directory's values.
+	if len(u.Scheme) == 0 {
+		u.Scheme = wd.Scheme
```

Comment:
Can you do the same thing for the Port as well for completeness? In case someone puts the HTTP server on the non-default port.

Reply:
The port is part of u.Host